core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1025,7 +1036,6 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean) extends Abstra
     require(logCleanerDedupeBufferSize / logCleanerThreads > 1024 * 1024, "log.cleaner.dedupe.buffer.size must be at least 1MB per cleaner thread.")
     require(replicaFetchWaitMaxMs <= replicaSocketTimeoutMs, "replica.socket.timeout.ms should always be at least replica.fetch.wait.max.ms" +
       " to prevent unnecessary socket timeouts")
-    require(replicaFetchMaxBytes >= messageMaxBytes, "replica.fetch.max.bytes should be equal or greater than message.max.bytes")
```

Comment:
I think this may make sense for cases where large messages are rare but can occur. For those cases, you may want to keep the default `replica.fetch.max.bytes` while allowing for larger messages. What do you think? It would also be good to get @junrao's input on this one.